kernel/userspace.c

```diff
@@ -222,7 +226,8 @@ void k_object_access_all_grant(void *object)
 	}
 }
 
-int _k_object_validate(struct _k_object *ko, enum k_objects otype, int init)
+int _k_object_validate(struct _k_object *ko, enum k_objects otype,
+		       enum _obj_init_check init)
 {
 	if (!ko || (otype != K_OBJ_ANY && ko->type != otype)) {
```

Comment:
Unrelated to the patch, but we might want to add `unlikely()` here for the error cases.